packages/flutter/lib/src/rendering/error.dart

```diff
@@ -157,11 +157,11 @@ class RenderErrorBox extends RenderBox {
           width -= padding.left + padding.right;
           left += padding.left;
         }
-        _paragraph!.layout(ui.ParagraphConstraints(width: width));
-        if (size.height > padding.top + _paragraph!.height + padding.bottom) {
+        _paragraph.layout(ui.ParagraphConstraints(width: width));
```

Comment:
The web compiler is unhappy about the removal of `!` here and below:

```
/b/s/w/ir/x/w/flutter/bin/cache/dart-sdk/bin/dart --disable-dart-dev /b/s/w/ir/x/w/flutter/bin/cache/dart-sdk/bin/snapshots/frontend_server.dart.snapshot --sdk-root /b/s/w/ir/x/w/flutter/bin/cache/flutter_web_sdk/ --incremental --target=dartdevc --experimental-emit-debug-metadata -DDART_HHH_BOT=false -DFLUTTER_WEB_AUTO_DETECT=false -DFLUTTER_WEB_USE_SKIA=true --output-dill /b/s/w/ir/x/t/flutter_tools.VOZITK/flutter_test.PPUCHG/out --packages /b/s/w/ir/x/w/flutter/packages/flutter/.dart_tool/package_config.json -Ddart.vm.profile=false -Ddart.vm.product=false --enable-asserts --track-widget-creation --filesystem-root /b/s/w/ir/x/w/flutter/packages/flutter/test --filesystem-root /b/s/w/ir/x/t/flutter_tools.VOZITK/flutter_test.PPUCHG/ --filesystem-scheme org-dartlang-app --initialize-from-dill build/a2632b1a2d119a1f9ce340959afb28ff.cache.dill.track.dill --platform file:///b/s/w/ir/x/w/flutter/bin/cache/flutter_web_sdk/kernel/ddc_outline_sound.dill --verbosity=error --sound-null-safety
[  +16 ms] <- compile org-dartlang-app:///main.dart
[+11509 ms] lib/src/rendering/error.dart:160:20: Error: Method 'layout' cannot be called on 'Paragraph?' because it is potentially null.
[   +2 ms]  - 'Paragraph' is from 'dart:ui'.
[        ] Try calling using ?. instead.
[        ]         _paragraph.layout(ui.ParagraphConstraints(width: width));
[        ]                    ^^^^^^
[   +6 ms] lib/src/rendering/error.dart:161:52: Error: Property 'height' cannot be accessed on 'Paragraph?' because it is potentially null.
[        ]  - 'Paragraph' is from 'dart:ui'.
[        ] Try accessing using ?. instead.
[        ]         if (size.height > padding.top + _paragraph.height + padding.bottom) {
[        ]                                                    ^^^^^^
[        ] lib/src/rendering/error.dart:61:28: Context: '_paragraph' refers to a property so it couldn't be promoted.
[        ] See http://dart.dev/go/non-promo-property
[        ]   late final ui.Paragraph? _paragraph;
[        ]                            ^
[   +1 ms] lib/src/rendering/error.dart:164:38: Error: The argument type 'Paragraph?' can't be assigned to the parameter type 'Paragraph' because 'Paragraph?' is nullable and 'Paragraph' isn't.
[        ]  - 'Paragraph' is from 'dart:ui'.
[        ]         context.canvas.drawParagraph(_paragraph, offset + Offset(left, top));
```

(from https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8770206943273787297/+/u/run_test.dart_for_web_canvaskit_tests_shard_and_subshard_0/test_stdout)

Reply:
The `!` should be removable because of `if (_paragraph != null)` in line 152, though. This issue also only surfaces on the web. When running in the VM, the removal of the `!` is not causing any trouble. Presumably, there is a bug somewhere in the web compilation pipeline?

@stereotype441 Any idea what may be going on here? If necessary, I can try to boil this down to a smaller repro case.

Reply:
We likely have to wait for https://github.com/dart-lang/sdk/commit/1da590fc3f6af8dc505e926551c0986fc05a67b6 to roll into Flutter, which is expected to fix this problem.

Reply:
Fix rolled in with https://github.com/flutter/flutter/pull/134763.